src/Interpreters/ExecuteScalarSubqueriesVisitor.cpp

```diff
@@ -186,8 +187,17 @@ void ExecuteScalarSubqueriesMatcher::visit(const ASTSubquery & subquery, ASTPtr
         {
             auto io = interpreter->execute();
 
+            auto cancellation_checker = data.getContext()->getQueryCancellationChecker();
+            UInt64 interactive_delay = data.getContext()->getQueryInteractiveDelay();
             PullingAsyncPipelineExecutor executor(io.pipeline);
-            while (block.rows() == 0 && executor.pull(block));
+            while (block.rows() == 0 && executor.pull(block, interactive_delay))
```

Comment:
Here I guess it can be converted to `CompletedPipelineExecutor`, and then `QueryPipeline::setProgressCallback` should be used, but actually `CompletedPipelineExecutor` looks complicated enough, maybe it is better to add support of progress callback in `PullingAsyncPipelineExecutor`